cli/azd/pkg/commands/pipeline/pipeline.go

```diff
@@ -124,6 +124,28 @@ const (
 	envPersistedKey string = "AZD_PIPELINE_PROVIDER"
 )
 
+func resolveProvider(env *environment.Environment, projectPath string) (string, error) {
+	// 1) if provider is set on azure.yaml, it should override the `lastUsedProvider`, as it can be changed by customer
+	// at any moment.
+	prj, err := project.LoadProjectConfig(projectPath)
+	if err != nil {
+		return "", fmt.Errorf("finding pipeline provider: %w", err)
+	}
+	if prj.Pipeline.Provider != "" {
+		return prj.Pipeline.Provider, nil
+	}
+
+	// 2) check if there is a persisted value from a previous run in env
+	if lastUsedProvider, configExists := env.Values[envPersistedKey]; configExists {
+		// Setting override value based on last run. This will force detector to use the same
+		// configuration.
+		return lastUsedProvider, nil
+	}
```

Comment:
Do you think we could remove all this, in favor of asking user: "Do you want to save `github` as the default provider?" Then update `azure.yaml` to set the provider?

Reply:
if they say `no`... right.. that's where we would still need this...

Imagine I run with `azdo` and when prompt, I select `no`.
Then I run again and forgot to ask for azdo...  the default would give me `GitHub`...